### DISCUSSION TITLE
If possible, how do I use an animated gif in equi?

### DISCUSSION THREAD

**Discussion by ray-5050** [emilk/egui#1521] [Q&A]
I can set up egui to show an image but I would like to display a gif that is animated.  I use the same setup but only got the first frame.  Is this possible in egui.  Since its only about 10 seconds, I thought about splitting it up and then loading a new image with each cycle.  Thanks for your help.

**Comment by possdog** [comment]
I don't believe egui has built-in functionality for this, but I think it would be *relatively* easy to do yourself. Load each frame of the gif separately into memory, display one image at a time using egui, and use a simple timer to decide when to move to the next frame.

**Comment by ray-5050** [comment]
Yep, just what I did and it works for my purpose.  I used std::fs::read() so that I didn't have to use the literal argument in include_bytes!().

**Comment by zdimension** [comment]
Support for animated GIFs has been added in https://github.com/emilk/egui/pull/4620.
